File: solutions/c/square-root/1/square_root.c
#include "square_root.h"
#include <stdint.h>
unsigned int square_root(unsigned int num) {
    float number = (float)num;
    const float x2 = number * 0.5F;
	const float threehalfs = 1.5F;

	union {
		float f;
		uint32_t i;
	} conv = {number};
	conv.i = 0x5f3759df - ( conv.i >> 1 );
	conv.f *= threehalfs - x2 * conv.f * conv.f;
	return (unsigned int)(1 / conv.f);
}

File: solutions/go/square-root/1/square_root.go
package squareroot

import (
    "math"
    "errors"
)

func SquareRoot(number int) (int, error) {
    if number < 0 {
        return -1, errors.New("Negative value")
    }
	num := float32(number)
    x2 := num * 0.5
    threehalfs := float32(1.5)

    i := math.Float32bits(num)
    i = 0x5f3759df - (i >> 1)
    f := math.Float32frombits(i)
    f = f * (threehalfs - (x2 * f * f))

    return int(1.0 / f), nil
}
